codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsJsonRpc1_0.java

```diff
@@ -36,4 +45,77 @@ protected String getDocumentContentType() {
     public ShapeId getProtocol() {
         return AwsJson1_0Trait.ID;
     }
+
+    @Override
+    protected Set<StructureShape> generateErrorShapes(
+        GenerationContext context, OperationShape operation, Symbol responseType) {
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors, (writer) -> AwsJsonRpc1_0.awsQueryCompatibleDefaultBlockWriter(writer));
+        } else {
+            return HttpProtocolGeneratorUtils.generateErrorDispatcher(
+                context, operation, responseType, this::writeErrorMessageCodeDeserializer,
+                this::getOperationErrors);
+        }
+    }
+
+    @Override
+    protected void deserializeError(GenerationContext context, StructureShape shape) {
+        GoWriter writer = context.getWriter().get();
+        Symbol symbol = context.getSymbolProvider().toSymbol(shape);
+        String functionName = ProtocolGenerator.getDocumentDeserializerFunctionName(shape, context.getService(),
+                getProtocolName());
+
+        AwsProtocolUtils.initializeJsonDecoder(writer, "errorBody");
+        AwsProtocolUtils.decodeJsonIntoInterface(writer, "");
+        writer.write("output := &$T{}", symbol);
+        writer.write("err := $L(&output, shape)", functionName);
+        writer.write("");
+        AwsProtocolUtils.handleDecodeError(writer);
+        writer.write("errorBody.Seek(0, io.SeekStart)");
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            writer.write("awsQueryErrorCode := getAwsQueryErrorCode(response)");
+            writer.openBlock("if awsQueryErrorCode != \"\" {", "}", () -> {
+                writer.write("output.ErrorCodeOverride = &awsQueryErrorCode");
+            });
+        }
+        writer.write("return output");
+    }
+
+    @Override
+    public void generateSharedDeserializerComponents(GenerationContext context) {
+        super.generateSharedDeserializerComponents(context);
+        if (isAwsQueryCompatibleTraitFound(context)) {
+            GoWriter writer = context.getWriter().get();
```

Comment:
Import `SMITHY_HTTP_TRANSPORT` (will also need to import package `SmithyGoDependency`):

```suggestion
        if (isAwsQueryCompatibleTraitFound(context)) {
            writer.addUseImports(SmithyGoDependency. SMITHY_HTTP_TRANSPORT);
            GoWriter writer = context.getWriter().get();
```